bundle/direct/dresources/secret_scope_acls.go

```diff
@@ -0,0 +1,223 @@
+	}
+	return scopeName, nil, nil
+}
+
+func (r *ResourceSecretScopeAcls) DoUpdate(ctx context.Context, id string, state *SecretScopeAclsState, _ *Changes) (*SecretScopeAclsState, error) {
+	_, err := r.setACLs(ctx, state.ScopeName, state.Acls)
+	return nil, err
+}
+
+func (r *ResourceSecretScopeAcls) DoUpdateWithID(ctx context.Context, oldID string, state *SecretScopeAclsState) (string, *SecretScopeAclsState, error) {
+	// Use state.ScopeName instead of oldID because when the parent scope is recreated,
+	// state.ScopeName will have the new (resolved) scope name, while oldID still has the old name
+	scopeName, err := r.setACLs(ctx, state.ScopeName, state.Acls)
+	if err != nil {
+		return "", nil, err
+	}
+	return scopeName, nil, nil
+}
+
+func (r *ResourceSecretScopeAcls) FieldTriggers(_ bool) map[string]deployplan.ActionType {
+	return map[string]deployplan.ActionType{
+		"scope_name": deployplan.ActionTypeUpdateWithID, // When scope name changes, ID changes
+		"acls":       deployplan.ActionTypeUpdate,       // When ACLs change, just update
+	}
+}
+
+// Removing ACLs is a no-op, to match the behavior for permissions and grants.
+func (r *ResourceSecretScopeAcls) DoDelete(ctx context.Context, id string) error {
+	return nil
+}
+
+// setACLs reconciles the desired ACLs with the current state
+func (r *ResourceSecretScopeAcls) setACLs(ctx context.Context, scopeName string, desiredAcls []workspace.AclItem) (string, error) {
+	// Get current ACLs
+	currentAcls, err := r.client.Secrets.ListAclsAll(ctx, workspace.ListAclsRequest{
+		Scope: scopeName,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to list current ACLs: %w", err)
+	}
+
+	// Build maps for reconciliation
+	desired := make(map[string]workspace.AclPermission)
+	for _, perm := range desiredAcls {
+		desired[perm.Principal] = perm.Permission
+	}
+
+	current := make(map[string]workspace.AclPermission)
+	for _, acl := range currentAcls {
+		current[acl.Principal] = acl.Permission
+	}
+
+	// Collect operations to perform in parallel
+	var toSet []workspace.PutAcl
+	var toDelete []workspace.DeleteAcl
+
+	// Find ACLs to set (new or changed)
+	for principal, permission := range desired {
+		if currentPerm, exists := current[principal]; !exists || currentPerm != permission {
+			toSet = append(toSet, workspace.PutAcl{
+				Scope:      scopeName,
+				Principal:  principal,
+				Permission: permission,
+			})
+		}
+	}
+
+	// Find ACLs to delete
+	for principal := range current {
+		if _, exists := desired[principal]; !exists {
+			toDelete = append(toDelete, workspace.DeleteAcl{
+				Scope:     scopeName,
+				Principal: principal,
+			})
+		}
+	}
+
+	// Execute all operations in parallel using errgroup
+	g, ctx := errgroup.WithContext(ctx)
+
+	// Set ACLs in parallel
+	for _, acl := range toSet {
+		g.Go(func() error {
+			if err := r.client.Secrets.PutAcl(ctx, acl); err != nil {
+				return fmt.Errorf("failed to set ACL %v for principal %q: %w", acl, acl.Principal, err)
```

Comment:
returning an error will cancel other requests in flight; should they be cancelled?

It seems safer is to collect all errors